buildSrc/src/main/java/org/elasticsearch/gradle/util/ports/AvailablePortAllocator.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.util.ports;
+
+import org.gradle.internal.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class AvailablePortAllocator implements PortAllocator {
+    public static final int MIN_PRIVATE_PORT = 49152;
+    public static final int MAX_PRIVATE_PORT = 65535;
+    public static final int DEFAULT_RANGE_SIZE = 100;
+
+    private final List<ReservedPortRange> reservations = new ArrayList<ReservedPortRange>();
+
+    private final Lock lock = new ReentrantLock();
+
+    private ReservedPortRangeFactory portRangeFactory = new DefaultReservedPortRangeFactory();
+
+    @Override
+    public int assignPort() {
```

Comment:
From what I can tell `assignePort` and `releasePort` are never used. Is there something that needs to be still wired up?

Reply:
kind of an oversight. I remove this as it's not used at the moment